### PR TITLE
Merge Collator branch into develop

### PR DESCRIPTION
**Pull request by jamesray1**
Creating a PR to start the review process.

### CODE REVIEW

File: src/collator.rs [hunk 1]
```diff
@@ -0,0 +1,99 @@
+use collation;
+use message;
+
+use std::sync::mpsc;
+use std::collections::HashMap;
+
+pub struct Collator {
```
Comment (jamesray1) [inline src/collator.rs]: What's the rationale for the design of this Struct?

From the infographic: periodically checks SMC status (particularly for the outcome of `get_eligible_collator()`.

Then in the SMC,`get_eligible_collator()` returns the `address` of the eligible collator.

File: src/collator.rs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    listener: mpsc::Receiver<message::Message>,
+}
+
+impl Collator {
+    pub fn new(listener: mpsc::Receiver<message::Message>) -> Collator {
+        Collator {
+            eligible: false,
+            shard_id: 0,
+            listener,
+            collation_trees: HashMap::new()
```
Comment (jamesray1) [inline src/collator.rs]: It's best to keep the order the same, you have collation_trees and listener swapped around.

**Comment by ChosunOne** [comment]
I haven't intended to merge this yet since I don't yet have full test coverage.

**Comment by jamesray1** [comment]
I wanted to open the PR so that I could start the review process. But I can see that the build is failing, and the unit tests need to be fixed as they are all failing.

**Comment by jamesray1** [comment]
I suggest to halt work on this until a new spec is defined and work on the tasks in #13 in the meantime.

**Comment by ChosunOne** [comment]
This branch is deprecated.  Please see the new notary branch for further work on the notaries.

**Pull request closed without merging**